R/audits.R

```diff
@@ -1,110 +1,49 @@
-#'   `vanity` is eligible as a vanity url.
+#' @return logical indicating if the vanity URL is available.
 #'
 #' @family audit functions
 #' @export
-audit_vanity_urls <- function(apps, server_url, vanity = NULL) {
-  # TODO: why does vanities not work?
-  urls <- get_field(apps, "url")
-  parse_server <- httr::parse_url(server_url)
-  if (is.null(parse_server$scheme)) {
-    stop(glue::glue("ERROR: protocol (i.e. http:// or https://) not defined on server_url={server_url}"))
-  }
-  content <- sapply(urls, function(u) {
-    trim_vanity(u, parse_server$path)
-  })
+vanity_is_available <- function(connect, vanity) {
+  current_vanities <- connect$GET(v1_url("vanities"))
+  current_vanity_paths <- purrr::map_chr(current_vanities, "path")
 
-  vanities <- content[!grepl("content/\\d+", content)]
+  # In case a full URL has been given, prune it down to just the path
+  vanity <- sub(paste0("^", connect$server), "", vanity)
+  # and make sure it has a leading and trailing slash
+  vanity <- sub("^/?(.*[^/])/?$", "/\\1/", vanity)
 
-  if (!is.null(vanity)) {
-    return(
-      ifelse(sprintf("%s/", vanity) %in% vanities, sprintf("%s Not Available", vanity), sprintf("%s Available", vanity))
-    )
-  }
-  vanities
+  !(vanity %in% current_vanity_paths)
 }
 
-trim_vanity <- function(url, server_path) {
-  parsed_url <- httr::parse_url(url)
-  if (nchar(server_path) > 0) {
-    # remove the trailing slash
-    server_path <- base::sub("^(.*)/$", "\\1", server_path)
-    vanity <- sub(server_path, "", parsed_url$path)
-  } else {
-    vanity <- parsed_url$path
-  }
-
-  # ensure leading and trailing slash
-  base::sub("^/?(.*[^/])/?$", "/\\1/", vanity)
-}
-
-
 #' Audit R Versions
 #'
 #' \lifecycle{experimental}
 #'
-#' @param apps App list, see `cache_apps`
+#' @param content `data.frame` of content information, as from [get_content()]
 #'
 #' @return A plot that shows the R version used by content over time and in
 #'   aggregate.
 #' @family audit functions
 #' @export
-audit_r_versions <- function(apps) {
-  r_versions <- get_field(apps, "r_version", TRUE)
-  published <- get_field(apps, "last_deployed_time", TRUE)
+audit_r_versions <- function(content) {
+  if (!requireNamespace("ggplot2", quietly = TRUE)) {
+    stop("ggplot2 is required for this function")
+  }
+  if (!requireNamespace("gridExtra", quietly = TRUE)) {
+    stop("gridExtra is required for this function")
+  }
 
-  # TODO: this is not pretty
-  timeline <- data.frame(
-    stringsAsFactors = FALSE,
-    r_version = unlist(r_versions),
-    published = do.call(c, unname( # this flattens the list while preserving the date time
-      lapply(
-        published[!sapply(r_versions, is.null)], # filter out records w/o r version
-        function(d) {
-          lubridate::ymd_hms(d)
-        }
-      ) # convert to date time
-    ))
-  )
+  timeline <- content[!is.na(content$r_version), c("r_version", "last_deployed_time")]
```

Comment:
I'm fine to bump the lintr to 100 characters, but it would be good not to make it even more angry than it already is...

Reply:
I added a .lintr file to do that.